libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/model/LaneFactory.kt

```diff
@@ -0,0 +1,18 @@
+package com.mapbox.navigation.ui.maneuver.model
+
+import com.mapbox.navigation.base.ExperimentalMapboxNavigationAPI
+
+/**
+ * A factory exposed to build a [Lane] object.
+ */
+@ExperimentalMapboxNavigationAPI
+object LaneFactory {
+
+    /**
+     * Build [Lane] given appropriate arguments
+     */
+    fun buildLane(
```

Comment:
> Function "buildLane" is never used 

Why is this exposed? How's this going to be used? Same thing applies to the rest of the factories 👇

Reply:
There is a possibility that customers might want to create there own `Maneuver` object and append it to the list of `Maneuver`. They cannot do it today because `Maneuver` is an `internal class`. As we agreed, we don't want to make these classes public. Hence the only 2 options left is to either convert them to `Builder` or use a `Factory`.
The problem with `Builder` is we would have to assign default values or, if not possible, then make the properties nullable which is contradictory. So, the decision to use `Factory` as it falls inline with our refactor to internal factories for objects.
Hence, these factories are not used now, but they are exposed for customers wanting to create their own Maneuver objects and injecting them into the view.
cc @Guardiola31337

Reply:
Couldn't `Maneuver` adopt the `toMutableVaue()` / `toImmutableValue()` methodology as was outlined in our architecture document from earlier in the year? See the `RouteSetValue` class as an example.

Reply:
I don't see why we cannot. What do you think? @LukasPaczos

Reply:
Looking at more details and discussing internally with @cafesilencio , looks like we cannot use `toMutable` and `toImmutable` in this case. The idea for `Maneuvers` is that once you get a list of `Maneuver` from the API, as a developer you should be able to add a completely new `Maneuver` object to the list. This would require us to expose some kind of ability to create a new `Maneuver` object and add it.

Reply:
Is this something we really want to expose publicly? Wondering if we can wait until we hear about this feature request? If that was the case already, disregard this comment.

Reply:
yep we already heard about it.